### DISCUSSION TITLE
HotKey/shortcut on MenuItem with Click event

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7497] [Q&A]
I'm trying to put a "shortcut" on a menu item. The MenuItem has a click event, rather than a binding command.

It's not working and, moreover, I don't understand the different between a Gesture and HotKey. All examples I can find refer to a bind command. I'm coming from a Qt background -- Is there a "straightforward" ShortCut equivalent in Avalonia which will fire the Click event?

                <MenuItem Header="_Open Solution..."
                    HotKey="Ctrl+O"
                    InputGesture="Ctrl+O"
                    Click="OpenClickHandler"
                />

**Comment by amwx** [comment]
1. Hotkeys only work by executing a Command, they don't raise the click event

2. `Hotkey` registers the given key combination with the hotkey manager to let that keyboard shortcut execute the command
https://github.com/AvaloniaUI/Avalonia/blob/765f2048d321c0e4c4932788238930f88012376d/src/Avalonia.Controls/MenuItem.cs#L215-L222

`InputGesture` allows you to specify a key gesture for the menu item without registering at hotkey. Useful if the same Hotkey is used in multiple areas as you don't want to register the same hotkey on multiple elements (bad things happen). 

https://github.com/AvaloniaUI/Avalonia/blob/765f2048d321c0e4c4932788238930f88012376d/src/Avalonia.Controls/MenuItem.cs#L243-L254

In your case, if you wanted to handle the click event instead of using Commands, you could set the InputGesture to `Ctrl+O` and then listen in the Window KeyDown or KeyUp handlers for `Ctrl+O` and then execute your logic yourself.

**Comment by kuiperzone** [comment]
Thank you very much for the detailed answer. It helps a lot. Cheers.
